### PR TITLE
Move configuration to yaml format

### PR DESCRIPTION
**Pull request by javanlacerda**
Contribute towards https://github.com/sigstore/fulcio/issues/1111
#### Summary

Migrate the configuration file from json to yaml.
It also removes the fulcio-config.yaml file that isn't used. (I've double checked it by running the docker-compose. It worked fine)
It concentrate the issuers and meta-issuers in a single file that can be found at `config/identity/config.yaml`
It also removes the `https://auth-staging.eclipse.org/realms/sigstore` from the list of issuers, as it is unavailable.
It is necessary to improve the onboarding for new OIDC providers, and make it easier to migrate the CI providers to the new generic logic that has been developed here https://github.com/sigstore/fulcio/pull/1679

cc @haydentherapper

### CODE REVIEW

**Comment by haydentherapper** [comment]
Some suggestions:
* Have a single config.yaml file which contains all of the trusted providers
* Remove config/fulcio-config.yaml, double checking that it's not used
* Add the email and description to the configuration struct
* Omit email/description when outputting configuration via API

**Comment by javanlacerda** [comment]
> Some suggestions:
> 
> * Have a single config.yaml file which contains all of the trusted providers
> * Remove config/fulcio-config.yaml, double checking that it's not used
> * Add the email and description to the configuration struct
> * Omit email/description when outputting configuration via API

Thank you for the suggestions. I managed to address the two first in this PR and I intend to address the other two in a next PR.

File: config/identity/config.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    type: email
+  https://agent.buildkite.com:
+    issuer-url: https://agent.buildkite.com
+    client-id: sigstore
+    type: buildkite-job
+  https://allow.pub:
+    issuer-url: https://allow.pub
+    client-id: sigstore
+    type: spiffe
+    spiffe-trust-domain: allow.pub
```
Comment (haydentherapper) [inline config/identity/config.yaml:28]: You're missing the entry for `https://auth-staging.eclipse.org/realms/sigstore`

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -28,22 +28,16 @@ import (
 )
```
Comment (haydentherapper) [inline pkg/config/config_test.go:29]: Can we have a test for both yaml and json?

File: pkg/config/fulcio_config_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+// It tests that the config/fulcio-config.yaml is properly parsable
+func TestLoadFulcioConfig(t *testing.T) {
+	_, path, _, _ := runtime.Caller(0)
+	basepath := filepath.Dir(path)
+	b, err := os.ReadFile(basepath + "/../../config/identity/config.yaml")
+	if err != nil {
+		t.Errorf("read file: %v", err)
+	}
+
+	cfg := FulcioConfigMap{}
```
Comment (haydentherapper) [inline pkg/config/fulcio_config_test.go:42]: Do we need this config map, or can we just call `Read(b)`?

File: pkg/config/fulcio_config_test.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	"testing"
+
+	"gopkg.in/yaml.v3"
+)
+
+type FulcioConfigMap struct {
+	Data map[string]string `yaml:"data,omitempty"`
+}
+
+// It tests that the config/fulcio-config.yaml is properly parsable
```
Comment (haydentherapper) [inline pkg/config/fulcio_config_test.go:33]: Can we make this comment more precise, that this is a CI test for verifying that the configuration to be used by the public good instance is valid? (it's doing more than verifying that it's parseable)

**Review by haydentherapper — CHANGES_REQUESTED** [review]
Can you update the documentation for https://github.com/sigstore/fulcio/blob/1a404b1fe7deb4e322326e18ea3a8c4a4574e5ef/docs/oidc.md?plain=1#L14 and https://github.com/sigstore/fulcio/blob/1a404b1fe7deb4e322326e18ea3a8c4a4574e5ef/tools/loadtest/README.md?plain=1#L27 to update references to the deleted yaml?

**Comment by haydentherapper** [comment]
@cpanato can you review the updates around CI?

File: config/identity/config.yaml (hunk 1 above)
Comment (javanlacerda) [reply config/identity/config.yaml:28]: I removed it as it is not working anymore. this staging version is returning 404

Comment (haydentherapper) [reply config/identity/config.yaml:28]: Ah yep, I see your comment, sg.

File: pkg/config/config_test.go (hunk 2 above)
Comment (javanlacerda) [reply pkg/config/config_test.go:29]: sure! on it

File: pkg/config/fulcio_config_test.go (hunk 3 above)
Comment (javanlacerda) [reply pkg/config/fulcio_config_test.go:42]: we didn't! removing

File: pkg/config/fulcio_config_test.go (hunk 4 above)
Comment (javanlacerda) [reply pkg/config/fulcio_config_test.go:33]: yep! it makes a lot of sense. on it

**Comment by javanlacerda** [comment]
> Can you update the documentation for
> 
> https://github.com/sigstore/fulcio/blob/1a404b1fe7deb4e322326e18ea3a8c4a4574e5ef/docs/oidc.md?plain=1#L14
> 
> and
> https://github.com/sigstore/fulcio/blob/1a404b1fe7deb4e322326e18ea3a8c4a4574e5ef/tools/loadtest/README.md?plain=1#L27
> 
> to update references to the deleted yaml?

Done! I will address the other doc updates in the PR removing the federation stuff. make sense?

**Review by haydentherapper — APPROVED** [review]
Thanks! Will merge Monday if there's no more comments.

**Review by cpanato — APPROVED** [review]

**Pull request merged**